### PR TITLE
chore: update hooks-store PR

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Pulled the latest changes from `develop` into https://github.com/cowprotocol/cowswap/pull/4034

image

image

image

### CODE REVIEW

File: apps/cowswap-frontend/src/modules/hooks/dapps/BuildHookApp/gnosis-logo.svg
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/hooks/dapps/BuildHookApp/gnosis-logo.svg:1]: This file already exists https://github.com/cowprotocol/cowswap/blob/develop/apps/cow-fi/public/images/gnosis-chain.svg

File: apps/cowswap-frontend/src/modules/appData/updater/AppDataHooksUpdater.ts [hunk 1]
```diff
@@ -42,30 +47,31 @@ export function AppDataHooksUpdater(): null {
...
       // Update the hooks
-      console.log('[AppDataHooksUpdater]: Set hooks', hooks)
       updateAppDataHooks(hooks)
-      permitDataPrev.current = permitData
+      hooksPrev.current = hooks
     } else {
       // There was a hook data, but not anymore. The hook needs to be removed
-      console.log('[AppDataHooksUpdater] Clear hooks')
       updateAppDataHooks(undefined)
-      permitDataPrev.current = undefined
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/appData/updater/AppDataHooksUpdater.ts:66]: You have a really hawk eye!

It comes from this [PR](https://github.com/cowprotocol/cowswap/pull/4034), but I'm pretty sure this is a mistake. @anxolin right?

File: apps/cowswap-frontend/src/modules/hooks/dapps/BuildHookApp/gnosis-logo.svg
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooks/dapps/BuildHookApp/gnosis-logo.svg:1]: Replaced with @cowprotocol/assets/cow-swap/network-gnosis-chain-logo.svg, thanks!

File: apps/cowswap-frontend/src/modules/hooks/hookRegistry.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  name: 'Create Maker Vault',
+  description: 'Borrow DAI by creating a Vault',
+  type: HookDappType.IFRAME,
+  url: FAKE_URL,
+  image: daiImg,
+  version: FAKE_VERSION,
+}
+
+// const PRE_TEST = {
+//   name: 'Claim GNO ',
```
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooks/hookRegistry.tsx:53]: This code is from [the original PR](https://github.com/cowprotocol/cowswap/pull/4034), I would keep it as it and remove after refactoring I'm going to do in another iteration

File: apps/cowswap-frontend/src/modules/hooks/images/cowAMM.png
[the remark this answers is not in the document]
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/hooks/images/cowAMM.png:1]: It will be changed after the next refactoring

**Review by alfetopito — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/appData/updater/AppDataHooksUpdater.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply apps/cowswap-frontend/src/modules/appData/updater/AppDataHooksUpdater.ts:66]: Ups, that looks like a mistake. But also, not sure `undefined` is right either. Isn't this an array now? if we need to clear hook we would need to empty

But I think this logic looks to me is from the "permit", and its intended to just delete the permit. What I think we have to do is delete any hook of type permit

**Review by anxolin — APPROVED** [review]
Nothing from this PR, but we will need to remember to use the right links for tenderly. I guess this will just force people to have an account and use their own free account. Maybe something to review:

image


I have an issue with the token selector, its possible my PR was not working great there, so happy we merge and iterate (as this will be hidden):

https://github.com/cowprotocol/cowswap/assets/2352112/ee5a2086-593d-423d-85c0-5f62defe62de


Anyways, I managed to place a SWAP with a hook, so I will approve and we can do small fixes in DEVELOP.

https://dev.explorer.cow.fi/gc/orders/0x0520043f5c8f642ddeefaa2648239b5681413686c5c1c11f9b3ec4241441b84179063d9173c09887d536924e2f6eadbabac099f566867f85?tab=overview

**Pull request merged**